src/utils.ts

```diff
@@ -29,7 +29,7 @@ export const isObjectWithProperties = (data: any): boolean => {
 
 // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
 export const isFunction = (functionToCheck: any): boolean => {
-  return functionToCheck && {}.toString.call(functionToCheck) === '[object Function]';
+  return functionToCheck && functionToCheck instanceof Function;
```

Comment:
Will this also handle GeneratorFunction?

https://stackoverflow.com/a/38510353

Reply:
Honestly, I didn't think about this... Also I don't remember any storage which may uses generators.

Reply:
So, I checked and it works same

Reply:
cool, approved!